<?xml version="1.0"?>

<launch>

  <arg name="nr_of_uavs"        default="0"/>
  <arg name="ugv_ns"            default="/ugv/"/>

  <node pkg="mipp_packages" type="joint_explorer_node" name="joint_explorer_node" output="screen">
    <!-- General -->
    <!-- Parameters -->
    <param name="nr_of_uavs"                        value="$(arg nr_of_uavs)"/>
    <param name="planner_rate"                      value="100.0"/>   <!-- Hz -->
    <param name="planner_max_time"                  value="3.0"/>     <!-- s -->
    <param name="planner_max_ray_distance"          value="2.0"/>     <!-- m -->
    <param name="planner_unmapped_is_collision"     value="true"/>
    <!-- UGV -->  
    <!-- Parameters --> 
    <param name="planner_world_frame"               value="world"/>
    <param name="use_dynamic_range"                 value="true"/>
    <param name="dynamic_range_padding"             value="2.0"/>
    <param name="x_range_min"                       value="-10.0"/>
    <param name="x_range_max"                       value="10.0"/>
    <param name="y_range_min"                       value="-10.0"/>
    <param name="y_range_max"                       value="10.0"/>
    <param name="planner_min_distance_to_frontier"  value="4.0"/>
    <param name="planner_max_distance_to_frontier"  value="100.0"/>
    <param name="collision_threshold"               value="0"/>
    <!-- Remap topics -->
    <!-- * Published topics -->
    <!-- * Subscribed topics -->
    <!-- UAVs -->
    <!-- Parameters -->
    <param name="uav_world_frame"                   value="world"/>
    <param name="uav_takeoff_z"                     value="2.0"/>
    <param name="uav_camera_width"                  value="6.0"/>
    <param name="uav_camera_height"                 value="3.0"/>
    <param name="uav_camera_hfov"                   value="1.02974"/>
    <param name="uav_camera_ray_resolution"         value="1.0"/>
    <param name="uav_camera_range"                  value="7.5"/>
    <param name="planner_sample_radius"             value="10.0"/>
    <param name="planner_max_neighbor_distance"     value="2.0"/>
    <param name="planner_max_neighbor_yaw"          value="1.5708"/>
    <!-- Remap topics -->
    <!-- * Published topics -->
    <!-- * Subscribed topics -->
  </node>
</launch>

